### PR TITLE
Remove more complex heroku deployment example

### PR DESCRIPTION
**Pull request by smart-alek**
On Tuesday, @digitalmaster and @technomancy attempted to deploy to Heroku using our documentation. This PR is a result of their feedback.

Resolves [this JIRA ticket](https://circleci.atlassian.net/secure/RapidBoard.jspa?rapidView=67&modal=detail&selectedIssue=CIRCLE-10297).

### CODE REVIEW

File: jekyll/_cci2/deployment-integrations.md [hunk 1]
```diff
@@ -142,137 +142,50 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
+1. Create a Heroku account
+and [follow the instructions](https://devcenter.heroku.com/start)
+to set up a project in your chosen language.
 
-To connect to the Heroku Git server from CircleCI,
-create a new SSH key without a passphrase.
-The private key is added through the SSH Permissions page
-with a hostname of `git.heroku.com` as shown in the following image:
+2. Add the name of your Heroku application and your Heroku API key as [environment variables]({{ site.baseurl }}/2.0/env-vars/#adding-project-level-environment-variables).
+In this example, these variables are defined as `HEROKU_API_KEY` and `HEROKU_APP_NAME`, respectively.
```
Comment (technomancy) [inline jekyll/_cci2/deployment-integrations.md]: Wups; I think you got the order of the variables swapped here.

File: jekyll/_cci2/deployment-integrations.md [hunk 2]
```diff
@@ -142,137 +142,50 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
 
-![Add SSH Key]({{ site.baseurl }}/assets/img/docs/walkthrough6.png)
+3. [Create a new SSH key](https://devcenter.heroku.com/articles/keys) without a passphrase,
+and [add the key to Heroku](https://devcenter.heroku.com/articles/keys#adding-keys-to-heroku).
 
-Note the private key's fingerprint for later reference.
-Add the public key to Heroku on the [Account page](https://dashboard.heroku.com/account).
+4. To connect to Heroku,
+[add the SSH key to CircleCI](https://circleci.com/docs/2.0/add-ssh-key/) with a **Hostname** of `git.heroku.com`.
```
Comment (technomancy) [inline jekyll/_cci2/deployment-integrations.md]: Steps 3 and 4 are only needed when pushing over SSH. In the instructions below we push over https using `HEROKU_API_KEY` as the password, so the SSH key is not needed.

File: jekyll/_cci2/deployment-integrations.md [hunk 3]
```diff
@@ -142,137 +142,50 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
 
-Finally, update your `config.yml` file with a job and workflow section similar to the following:
+5. In your `.circleci/config.yml`,
+create a `deploy-job`,
+`checkout` your code,
+and add a command
+to deploy the master branch to Heroku.
```
Comment (technomancy) [inline jekyll/_cci2/deployment-integrations.md]: Just to make this crystal-clear, I'd make this: `to deploy the master branch to Heroku via git.`

File: jekyll/_cci2/deployment-integrations.md (hunk 2 above)
Comment (smart-alek) [reply jekyll/_cci2/deployment-integrations.md]: Do we lose anything by not documenting the SSH approach? Or are we comfortable with a firm stance on pushing over https?

File: jekyll/_cci2/deployment-integrations.md [hunk 4]
```diff
@@ -142,137 +142,44 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
+To configure CircleCI
+to deploy your application to Heroku,
+follow the steps below.
 
-To connect to the Heroku Git server from CircleCI,
-create a new SSH key without a passphrase.
-The private key is added through the SSH Permissions page
-with a hostname of `git.heroku.com` as shown in the following image:
+1. Create a Heroku account
+and [follow the instructions](https://devcenter.heroku.com/start)
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment-integrations.md]: and follow the instructions-> and follow the [Getting Started on Heroku] documentation

File: jekyll/_cci2/deployment-integrations.md [hunk 5]
```diff
@@ -142,137 +142,44 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
+to set up a project in your chosen language.
 
-![Add SSH Key]({{ site.baseurl }}/assets/img/docs/walkthrough6.png)
+2. Add the name of your Heroku application and your Heroku API key as [environment variables]({{ site.baseurl }}/2.0/env-vars/#adding-project-level-environment-variables).
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment-integrations.md]: as environment variables -> as environment variables, see the [Adding Project Environment Variables] section of the Using Environment Variables document for instructions.

File: jekyll/_cci2/deployment-integrations.md [hunk 6]
```diff
@@ -142,137 +142,44 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
 
-Note the private key's fingerprint for later reference.
-Add the public key to Heroku on the [Account page](https://dashboard.heroku.com/account).
-
-Finally, update your `config.yml` file with a job and workflow section similar to the following:
+4. In your `.circleci/config.yml`,
+create a `deploy-job`,
+`checkout` your code,
+and add a command
+to deploy the master branch to Heroku via git.
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment-integrations.md]: via-> using (using is more clear for non-English speakers than via)

File: jekyll/_cci2/deployment-integrations.md [hunk 7]
```diff
@@ -142,137 +142,44 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
 
 ## Heroku
 
-The built-in Heroku integration through the CircleCI UI is not implemented for CircleCI 2.0.
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment-integrations.md]: Awesome, glad to be getting rid of this!

File: jekyll/_cci2/deployment-integrations.md [hunk 8]
```diff
@@ -142,137 +142,44 @@ To deploy to Azure, use a similar job to the above example that uses an appropri
...
-            git push --force https://heroku:$HEROKU_API_KEY@git.heroku.com/$HEROKU_APP_NAME.git HEAD:refs/heads/master
-            sleep 5  # sleep for 5 seconds to wait for dynos
-            heroku restart
 workflows:
   version: 2
   build-deploy:
     jobs:
       - build-job
-      - deploy-jobs:  # only deploy when master successfully builds
+      - deploy-job:
```
Comment (michelle-luna) [inline jekyll/_cci2/deployment-integrations.md]: In the next section, we should change 'for additional details' or remove that link entirely (and if that documentation is incorrect/not what we should tell people, we should remove it from the python flask demo app as well and replace it with the easy way). All of this info being removed is still in the Project Walkthrough.

File: jekyll/_cci2/deployment-integrations.md (hunk 6 above)
Comment (smart-alek) [reply jekyll/_cci2/deployment-integrations.md]: 👍 Adding preference to style guide.

File: jekyll/_cci2/deployment-integrations.md (hunk 8 above)
Comment (smart-alek) [reply jekyll/_cci2/deployment-integrations.md]: Awesome. I'll remove this link for now and make a ticket to update the Project Walkthrough with the easier Heroku deployment method.

File: jekyll/_cci2/deployment-integrations.md (hunk 2 above)
Comment (technomancy) [reply jekyll/_cci2/deployment-integrations.md]: IMO the SSH method is much better for deploys done directly by a user, but the https+API-key-based method is better for unattended deploys.

**Comment by technomancy** [comment]
https://media.giphy.com/media/Q2LRWdJDq9xy8/giphy.gif

Let's get this merged.

**Pull request merged**